app/main.py

```
@@ -1,52 +1,32 @@
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.display import ReverseDisplay, ConsoleDisplay
+from app.serializers import JsonSerializer, XmlSerializer
+from app.print import ConsolePrint, ReversePrint
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    methods = {
+        "display": {
+            "reverse": ReverseDisplay,
+            "console": ConsoleDisplay,
+        },
+        "print": {
+            "console": ConsolePrint,
+            "reverse": ReversePrint
+        },
+        "serialize": {
+            "json": JsonSerializer,
+            "xml": XmlSerializer,
+        }
+    }
+
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            methods[cmd][method_type]().display(book)
         elif cmd == "print":
-            book.print_book(method_type)
+            methods[cmd][method_type]().print_book(book)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            return methods[cmd][method_type]().serialize(book)
```

these conditions are redundant, you already have a mapper. Also you should raise an error according to the original code